test/unit/DCAStrategies/dca-strategies-permissions-handler.spec.ts

```diff
@@ -51,6 +51,35 @@ contract('DCAStrategiesPermissionsHandler', () => {
       then('mint counter starts at 0', async () => {
         expect(await DCAStrategiesPermissionsHandlerMock.mintCounter()).to.equal(0);
       });
+      then('domain separator is the expected', async () => {
+        expect(await DCAStrategiesPermissionsHandlerMock.DOMAIN_SEPARATOR()).to.equal(
+          await domainSeparator(NFT_NAME, '1', chainId, DCAStrategiesPermissionsHandlerMock.address)
+        );
+      });
+    });
+  });
+
+  describe('totalSupply', () => {
+    when('no tokens have been minted ', () => {
+      then('supply is zero', async () => {
+        expect(await DCAStrategiesPermissionsHandlerMock.totalSupply()).to.equal(BigNumber.from(0));
+      });
+    });
+
+    when('some tokens have been minted and burned ', () => {
```

Comment:
I would split this test into two:
1. One test mints a position, and we check that `totalSupply` is increased to 1
2. Another tests mints a position, and then burns it. Then we check that `totalSupply` is 0

By doing this, you can test the behaviours separately and you are that they work correctly